src/neo4j_genai/experimental/components/text_splitters/llamaindex.py

```diff
@@ -54,8 +54,9 @@ async def run(self, text: str) -> TextChunks:
         Returns:
             TextChunks: The text split into chunks.
         """
+        chunks = self.text_splitter.split_text(text)
         return TextChunks(
             chunks=[
-                TextChunk(text=chunk) for chunk in self.text_splitter.split_text(text)
+                TextChunk(text=chunk, index=index) for index, chunk in enumerate(chunks)
```

Comment:
This might be a minor point but do you know for sure that the `split_text` method always returns chunks in the correct order for all LangChain and LlamaIndex splitters? I think LlamaIndex splitters have a `split_text_metadata_aware` method which includes an index which could be safer to use, and LangChain splitters have a similar method.

Reply:
Good point, I'll check what's in LC/LI, it's important to have the right chunk order.

Reply:
Yeah I'm pretty sure we're fine here. I would just double check as it's been a while since I looked into the text splitter classes for both libraries and I could be wrong

Reply:
I think we can assume it's OK. All the other methods I've seen are actually calling the `split_text` method and returning the same results with different format but without reordering of any kind.